pkg/csi/service/wcp/controller.go

```diff
@@ -1336,8 +1337,27 @@ func (c *controller) ControllerUnpublishVolume(ctx context.Context, req *csi.Con
 							}
 							nodeNames := nodesForVolume[req.VolumeId]
 							if len(nodeNames) == 1 && nodeNames[0] == req.NodeId {
-								log.Errorf("volume %q is still attached to node %q and podVM %q with moId %q", req.VolumeId,
-									req.NodeId, podVM.Reference().String(), podVM.Reference().Value)
+								log.Errorf("volume %q is still attached to node %q and podVM %q", req.VolumeId,
+									req.NodeId, podVM.Reference().String())
+								podvmpowerstate, err := podVM.PowerState(ctx)
+								if err != nil {
+									log.Errorf("failed to check the power state of pod vm: %q, error: %v",
+										podVM.Reference(), err)
+									return nil, csifault.CSIInternalFault, err
+								}
+								log.Infof("power state of pod vm: %q is %q", podVM.Reference(), podvmpowerstate)
+								if podvmpowerstate == types.VirtualMachinePowerStatePoweredOff {
+									log.Infof("attempting to detach volume %q from "+
```

Comment:
Can be debugf as the below return paths have the log statements

Reply:
done. changed to `log.Debugf`.